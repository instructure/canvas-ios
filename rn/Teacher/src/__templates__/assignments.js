//
// This file is part of Canvas.
// Copyright (C) 2017-present  Instructure, Inc.
//
// This program is free software: you can redistribute it and/or modify
// it under the terms of the GNU Affero General Public License as
// published by the Free Software Foundation, either version 3 of the
// License, or (at your option) any later version.
//
// This program is distributed in the hope that it will be useful,
// but WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
// GNU Affero General Public License for more details.
//
// You should have received a copy of the GNU Affero General Public License
// along with this program.  If not, see <https://www.gnu.org/licenses/>.
//

/* @flow */

import template, { type Template } from '../utils/template'

export const assignment: Template<Assignment> = template({
  id: '123456789',
  course_id: '987654321',
  name: 'learn to write code',
  description: 'a course that can take any beginner to professional in a matter of microseconds.  Harness the power of your favorite languages like PHP and Cold Fusion.',
  due_at: '2037-06-01T05:59:00Z',
  unlock_at: '2017-03-17T06:00:00Z',
  lock_at: '2037-06-01T05:59:59Z',
  created_at: '2017-03-17T19:15:25Z',
  updated_at: '2017-03-17T19:15:25Z',
  points_possible: 20,
  grading_type: 'points',
  assignment_group_id: '1376',
  peer_reviews: false,
  automatic_peer_reviews: false,
  position: 1,
  submission_types:
  [ 'online_text_entry',
    'online_url',
    'media_recording',
    'online_upload' ],
  needs_grading_count: 0,
  html_url: 'http://hostname.tld/whatever',
  overrides: [],
  grade_group_students_individually: false,
  group_category_id: null,
  has_overrides: false,
  only_visible_to_overrides: false,
  published: true,
  unpublishable: false,
  rubric_assessment: {},
  rubric_settings: null,
  rubric: null,
  anonymize_students: false,
})

export const assignmentGroup: Template<AssignmentGroup> = template({
  id: '1',
  name: 'Learn React Native',
  position: 1,
  assignments: [assignment({ assignment_group_id: '1' })],
  group_weight: 1,
  integration_data: {},
  sis_source_id: '',
})

export const assignmentDueDate: Template<AssignmentDate> = template({
  title: 'due',
  due_at: '2037-06-01T05:59:00Z',
  unlock_at: '2017-03-17T06:00:00Z',
  lock_at: '2037-06-01T05:59:59Z',
})

export const assignmentOverride: Template<AssignmentOverride> = template({
  id: '1',
  assignment_id: '1',
  student_ids: [],
  group_id: '1',
  course_section_id: '1',
  title: 'override',
  all_day: false,
  all_date_date: '',
  due_at: '2037-06-01T05:59:00Z',
  unlock_at: '2017-03-17T06:00:00Z',
  lock_at: '2037-06-01T05:59:59Z',
})
