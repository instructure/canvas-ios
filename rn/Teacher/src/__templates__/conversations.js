//
// This file is part of Canvas.
// Copyright (C) 2017-present  Instructure, Inc.
//
// This program is free software: you can redistribute it and/or modify
// it under the terms of the GNU Affero General Public License as
// published by the Free Software Foundation, either version 3 of the
// License, or (at your option) any later version.
//
// This program is distributed in the hope that it will be useful,
// but WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
// GNU Affero General Public License for more details.
//
// You should have received a copy of the GNU Affero General Public License
// along with this program.  If not, see <https://www.gnu.org/licenses/>.
//

/* @flow */

import template, { type Template } from '../utils/template'

export const conversation: Template<Conversation> = template({
  id: '123456789',
  subject: 'learn to write code',
  workflow_state: 'read',
  last_message: 'this is the worst class ever',
  last_message_at: '2037-06-01T05:59:00Z',
  last_authored_message: 'this is the worst class ever',
  last_authored_message_at: '2037-06-01T05:59:00Z',
  participants: [
    {
      id: '1234',
      name: 'artorias of the abyss',
    },
    {
      id: '6789',
      name: 'solaire of astora',
    },
  ],
  avatar_url: 'http://www.fillmurray.com/200/300',
  context_name: 'Test Class',
  context_code: 'course_1',
  audience_contexts: {},
  audience: [],
  message_count: 2,
  private: false,
  properties: null,
  starred: false,
  subscribed: false,
  visible: true,
})

export const conversationMessage: Template<ConversationMessage> = template({
  id: '123456789',
  body: 'this is neat',
  author_id: '1234',
  created_at: '2037-06-01T05:59:00Z',
  attachments: [
    {
      id: '1',
      display_name: 'something.pdf',
      mime_class: 'pdf',
    },
  ],
  generated: false,
})
