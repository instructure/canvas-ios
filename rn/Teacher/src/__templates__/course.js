//
// This file is part of Canvas.
// Copyright (C) 2017-present  Instructure, Inc.
//
// This program is free software: you can redistribute it and/or modify
// it under the terms of the GNU Affero General Public License as
// published by the Free Software Foundation, either version 3 of the
// License, or (at your option) any later version.
//
// This program is distributed in the hope that it will be useful,
// but WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
// GNU Affero General Public License for more details.
//
// You should have received a copy of the GNU Affero General Public License
// along with this program.  If not, see <https://www.gnu.org/licenses/>.
//

/* @flow */

import { CourseModel } from '../canvas-api/model'
import template, { type Template } from '../utils/template'
import { section } from './section'

export const courseTerm: Template<CourseTerm> = template({
  id: '1',
  created_at: '2017-03-17T19:15:25Z',
  name: 'Default Term',
})

export const course: Template<Course> = template({
  id: '1',
  account_id: '1',
  name: 'Learn React Native',
  short_name: 'rn',
  course_code: 'rn 101',
  image_download_url: 'https://farm3.staticflickr.com/2926/14690771011_945f91045a.jpg',
  is_favorite: true,
  default_view: 'wiki',
  enrollments: [{
    enrollment_state: 'active',
    role: 'TeacherEnrollment',
    role_id: '1',
    type: 'teacher',
    user_id: '1',
  }],
  permissions: {
    viewAnalytics: true,
    viewAllGrades: true,
  },
  workflow_state: 'available',
  term: {
    id: '1',
    created_at: '2018-02-08T23:13:39Z',
    end_at: '2030-02-08T23:13:39Z',
    start_at: '2018-02-08T23:13:39Z',
    name: 'Default Term',
    workflow_state: 'available',
  },
  end_at: '2030-02-08T23:13:39Z',
})

export const courseModel: Template<CourseModel> = overrides =>
  Object.assign(new CourseModel(course()), overrides)

export const courseWithSection: Template<Course> = function (defaults) {
  return course({
    ...defaults,
    sections: [section()],
  })
}

export const customColors: Template<CustomColors> = template({
  custom_colors: {
    course_1: '#fff',
  },
})

export const courseSettings: Template<CourseSettings> = template({
  allow_student_forum_attachments: true,
})
