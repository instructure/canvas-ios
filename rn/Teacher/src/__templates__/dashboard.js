//
// This file is part of Canvas.
// Copyright (C) 2019-present  Instructure, Inc.
//
// This program is free software: you can redistribute it and/or modify
// it under the terms of the GNU Affero General Public License as
// published by the Free Software Foundation, either version 3 of the
// License, or (at your option) any later version.
//
// This program is distributed in the hope that it will be useful,
// but WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
// GNU Affero General Public License for more details.
//
// You should have received a copy of the GNU Affero General Public License
// along with this program.  If not, see <https://www.gnu.org/licenses/>.
//

// @flow

import template, { type Template } from '../utils/template'

export const dashboardCardLink: Template<DashboardCardLink> = template({
  css_class: 'assignments',
  hidden: null,
  icon: 'icon-assignment',
  label: 'Assignments',
  path: '/courses/1/assignments',
})

export const dashboardCard: Template<DashboardCard> = template({
  assetString: 'course_1',
  courseCode: 'MET-132',
  enrollmentType: 'StudentEnrollment',
  href: '/courses/1',
  id: '1',
  image: null,
  links: [dashboardCardLink()],
  longName: 'A good course - MET',
  originalName: 'A good course',
  position: null,
  shortName: 'A good course',
  subtitle: 'enrolled as: Student',
  term: null,
})
