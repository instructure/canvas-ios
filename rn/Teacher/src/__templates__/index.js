//
// This file is part of Canvas.
// Copyright (C) 2018-present  Instructure, Inc.
//
// This program is free software: you can redistribute it and/or modify
// it under the terms of the GNU Affero General Public License as
// published by the Free Software Foundation, either version 3 of the
// License, or (at your option) any later version.
//
// This program is distributed in the hope that it will be useful,
// but WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
// GNU Affero General Public License for more details.
//
// You should have received a copy of the GNU Affero General Public License
// along with this program.  If not, see <https://www.gnu.org/licenses/>.
//

// @flow

export * from '../redux/__templates__/app-state'
export * from './account-notification'
export * from './addressBook'
export * from './api'
export * from './assignments'
export * from './attachment'
export * from './conversations'
export * from './course'
export * from './discussion'
export * from './enrollments'
export * from './error'
export * from './external-tool'
export * from './file'
export * from './grading-periods'
export * from './group'
export * from './helm'
export * from './launch-definitions'
export * from './mediaComment'
export * from './quiz'
export * from './quizSubmission'
export * from './role'
export * from './rubric'
export * from './section'
export * from './session'
export * from './submissions'
export * from './tab'
export * from './users'
export * from './help-links'
export * from './dashboard'
export * from '../modules/assignee-picker/__template__/Assignee'
