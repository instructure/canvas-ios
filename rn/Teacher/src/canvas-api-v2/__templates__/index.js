// @flow
export * from './submission'
export * from './user'
export * from './assignment'
export * from './course'
export * from './group'
export * from './section'
export * from './groupSet'

// QUERIES
export * from '../queries/__templates__/SubmissionList'
export * from '../queries/__templates__/ContextCard'
