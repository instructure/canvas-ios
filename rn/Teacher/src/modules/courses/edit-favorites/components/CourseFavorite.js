//
// This file is part of Canvas.
// Copyright (C) 2017-present  Instructure, Inc.
//
// This program is free software: you can redistribute it and/or modify
// it under the terms of the GNU Affero General Public License as
// published by the Free Software Foundation, either version 3 of the
// License, or (at your option) any later version.
//
// This program is distributed in the hope that it will be useful,
// but WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
// GNU Affero General Public License for more details.
//
// You should have received a copy of the GNU Affero General Public License
// along with this program.  If not, see <https://www.gnu.org/licenses/>.
//

// @flow

import React, { Component } from 'react'
import {
  TouchableHighlight,
  View,
  Image,
} from 'react-native'
import { colors, createStyleSheet } from '../../../../common/stylesheet'
import icon from '../../../../images/inst-icons'
import { Text } from '../../../../common/text'

type Props = {
  course: Course | Group,
  isFavorite: boolean,
  onPress: (contextID: string, favorited: boolean) => Promise<any>,
}

export default class CourseFavorite extends Component<Props> {
  onPress = () => {
    this.props.onPress(this.props.course.id, !this.props.isFavorite)
  }

  render () {
    return (
      <TouchableHighlight
        style={styles.courseRow}
        testID={'edit-favorites.course-favorite.' + this.props.course.id + (this.props.isFavorite ? '-favorited' : '-not-favorited') }
        onPress={this.onPress}
        accessibilityRole='button'
        accessibilityStates={ this.props.isFavorite ? [ 'selected' ] : []}
      >
        <View style={styles.courseWrapper}>
          <Image
            source={icon('star', this.props.isFavorite ? 'solid' : 'line')}
            style={[styles.favoritesImage, { tintColor: colors.primary }]}
          />
          <Text style={styles.courseName}>{this.props.course.name}</Text>
        </View>
      </TouchableHighlight>
    )
  }
}

const styles = createStyleSheet(colors => ({
  courseRow: {
    borderBottomWidth: 1,
    borderBottomColor: colors.borderMedium,
  },
  courseWrapper: {
    paddingHorizontal: 16,
    paddingVertical: 16,
    flexDirection: 'row',
    alignItems: 'center',
    backgroundColor: colors.backgroundLightest,
  },
  courseName: {
    marginRight: 16,
    fontWeight: '500',
  },
  favoritesImage: {
    marginRight: 10,
    width: 20,
    height: 20,
  },
}))
