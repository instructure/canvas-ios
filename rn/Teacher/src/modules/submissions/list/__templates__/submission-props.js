//
// This file is part of Canvas.
// Copyright (C) 2017-present  Instructure, Inc.
//
// This program is free software: you can redistribute it and/or modify
// it under the terms of the GNU Affero General Public License as
// published by the Free Software Foundation, either version 3 of the
// License, or (at your option) any later version.
//
// This program is distributed in the hope that it will be useful,
// but WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
// GNU Affero General Public License for more details.
//
// You should have received a copy of the GNU Affero General Public License
// along with this program.  If not, see <https://www.gnu.org/licenses/>.
//

// @flow

import template, { type Template } from '../../../../utils/template'
import type { SubmissionDataProps } from '../submission-prop-types'

const templates = {
  ...require('../../../../__templates__/submissions'),
}

const submissionWithHistory = templates.submissionHistory()

// $FlowFixMe
export const submissionProps: Template<SubmissionDataProps> = template({
  grade: 'B-',
  name: submissionWithHistory.user.name,
  status: 'submitted',
  userID: submissionWithHistory.user.id,
  avatarURL: submissionWithHistory.user.avatar_url,
  sectionID: '1',
})
