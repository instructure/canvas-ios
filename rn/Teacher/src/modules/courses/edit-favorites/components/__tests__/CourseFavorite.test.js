//
// This file is part of Canvas.
// Copyright (C) 2017-present  Instructure, Inc.
//
// This program is free software: you can redistribute it and/or modify
// it under the terms of the GNU Affero General Public License as
// published by the Free Software Foundation, either version 3 of the
// License, or (at your option) any later version.
//
// This program is distributed in the hope that it will be useful,
// but WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
// GNU Affero General Public License for more details.
//
// You should have received a copy of the GNU Affero General Public License
// along with this program.  If not, see <https://www.gnu.org/licenses/>.
//

// @flow

import 'react-native'
import { shallow } from 'enzyme'
import React from 'react'
import CourseFavorite from '../CourseFavorite'
import * as template from '../../../../../__templates__'
import icon from '../../../../../images/inst-icons'

let defaultProps = {
  id: '1',
  course: template.course(),
  isFavorite: true,
  onPress: () => Promise.resolve(),
}

describe('CourseFavorite', () => {
  it('renders favorited as selected', () => {
    const tree = shallow(<CourseFavorite {...defaultProps} />)
    expect(tree.prop('accessibilityRole')).toBe('button')
    expect(tree.prop('accessibilityStates')).toEqual([ 'selected' ])
    expect(tree.find('Image').prop('source')).toEqual(icon('star', 'solid'))
  })

  it('renders unfavorited without selected', () => {
    const tree = shallow(<CourseFavorite {...defaultProps} isFavorite={false} />)
    expect(tree.prop('accessibilityRole')).toBe('button')
    expect(tree.prop('accessibilityStates')).toEqual([])
    expect(tree.find('Image').prop('source')).toEqual(icon('star', 'line'))
  })

  it('calls props.onPress with the course id and the toggled favorite value', () => {
    const onPress = jest.fn()
    const tree = shallow(<CourseFavorite {...defaultProps} onPress={onPress} />)
    tree.simulate('Press')
    expect(onPress).toHaveBeenCalledWith(defaultProps.course.id, false)
  })
})
