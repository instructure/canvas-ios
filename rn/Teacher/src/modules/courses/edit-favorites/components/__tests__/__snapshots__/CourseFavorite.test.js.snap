// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders favorited correctly 1`] = `
<TouchableHighlight
  accessibilityTraits={
    Array [
      "selected",
      "button",
    ]
  }
  onPress={[Function]}
  style={
    Object {
      "borderBottomColor": "#e3e3e3",
      "borderBottomWidth": 1,
    }
  }
  testID="edit-favorites.course-favorite.1-favorited"
>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "white",
        "flexDirection": "row",
        "paddingHorizontal": 16,
        "paddingVertical": 16,
      }
    }
  >
    <Image
      source={
        Object {
          "testUri": "../../../src/images/star-filled.png",
        }
      }
      style={
        Array [
          Object {
            "height": 20,
            "marginRight": 10,
            "width": 20,
          },
          Object {
            "tintColor": "#374A59",
          },
        ]
      }
    />
    <Text
      style={
        Array [
          Object {
            "fontFamily": ".SFUIDisplay",
          },
          Object {
            "color": "#2D3B45",
            "fontSize": 16,
          },
          Object {
            "fontWeight": "500",
            "marginRight": 16,
          },
          Object {
            "fontFamily": ".SFUIDisplay-medium",
          },
        ]
      }
    >
      Learn React Native
    </Text>
  </View>
</TouchableHighlight>
`;

exports[`renders unfavorited correctly 1`] = `
<TouchableHighlight
  accessibilityTraits={
    Array [
      "button",
    ]
  }
  onPress={[Function]}
  style={
    Object {
      "borderBottomColor": "#e3e3e3",
      "borderBottomWidth": 1,
    }
  }
  testID="edit-favorites.course-favorite.1-not-favorited"
>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "white",
        "flexDirection": "row",
        "paddingHorizontal": 16,
        "paddingVertical": 16,
      }
    }
  >
    <Image
      source={
        Object {
          "testUri": "../../../src/images/star-lined.png",
        }
      }
      style={
        Array [
          Object {
            "height": 20,
            "marginRight": 10,
            "width": 20,
          },
          Object {
            "tintColor": "#374A59",
          },
        ]
      }
    />
    <Text
      style={
        Array [
          Object {
            "fontFamily": ".SFUIDisplay",
          },
          Object {
            "color": "#2D3B45",
            "fontSize": 16,
          },
          Object {
            "fontWeight": "500",
            "marginRight": 16,
          },
          Object {
            "fontFamily": ".SFUIDisplay-medium",
          },
        ]
      }
    >
      Learn React Native
    </Text>
  </View>
</TouchableHighlight>
`;
